### DISCUSSION TITLE
Home IDp with username/password flow

### DISCUSSION THREAD

**Discussion by lusnyaktah** [sventorben/keycloak-home-idp-discovery#461] [Q&A]
Hello, thanks for the fantastic library, it is very useful. 
I have approximately same requirement and going to use your concept but there is an additional requirement. I need always show username/password  with IDP's so user either will choose to continue by their IDP or username/password.

 Do you have any idea how I can achieve that ?

When I add username/password form in home_idp.ftl after submitting it again shows the same ftl

**Comment by sventorben** [comment]
Do you just want to show a list of IdPs below the username/password form fields? I don't really understand how that is different from what KC offers outbid the box. Maybe I am missing your point.

You cannot simply add fields in the ftl - you must also implement the logic behind it.

**Comment by sventorben** [comment]
Closing due to missing feedback.
